### ISSUE TITLE
Reviews

### ISSUE THREAD

**Issue by mathurinm** [computorg/published_202401_adrat_repulsion#1]
### 1st review 

The manuscript proposes an algorithm for the classification of configurations of points based on their ‘repulsiveness’. In a nutshell, the algorithm applies a classical learning algorithm (logistic regression and random forest) to features that measure the repulsiveness through some geometric characteristics (areas and perimeters) of the Voronoi cells of the corresponding configuration of points. The algorithm is inspired by some theoretical results in [1] that compare the Voronoi cells of the Ginibre ensemble (a point process with repulsion), to the Voronoi cells of a Poisson point process.
Unfortunately, I believe that this work does not meet the standards for publication in the journal for the following reasons:

> A formal description of the proposed algorithm is missing and some numerical simulations were not commented.
> The Python implementation lacks comments.

For a resubmission, I suggest the following changes:

- Adding a section with a formal description of the algorithm
- Reformulating some sentences (see the list below)
- Illustrating the proposed algorithm with other experiments. For example, what happens if we apply the algorithm to compare a poisson point process to a hardcore point process? How about the Ginibre ensemble vs a hardcore point process? …

[...]

[1] André Goldman. The Palm measure and the Voronoi tessellation for the Ginibre process. The Annals of Applied Probability, 20(1):90–128, February 2010. arXiv:math/0610243.

### 2nd review 

The manuscript has improved since the initial version. I think the authors should clarify the statistical framework they use, esp. in section "Machine Learning approach" and "Cartoradio data Tests". The writing seems to imply that they are designing a statistical test to detect repulsion but the way the test is constructed is very heuristic and offers no control over Type 1 error. A more accurate description is that the output of the classifier is just a composite score based on some statistics of the point process, tuned to discriminate between a Poisson process and a repulsive point process. I still wonder whether it wouldn't be simpler and more efficient to average some shape statistics over all cells except boundary cells. In addition, it'd be nice to at least discuss the fact that there are already plenty of summary statistics that are used to detect repulsion, the most common being estimators of the pair correlation function, and to state how they differ from the suggested approach. 

### First review
The paper makes an case for using the geometry of Voronoi cells to study repulsiveness in spatial point processes. My impression at this stage is that the paper falls a bit short of making a useful contribution but could be quite interesting after significant revisions.

There’s a huge literature on Voronoi diagrams of point sets, see for instance
ch. 9 in the classic textbook by Stoyan et al. (2013). It’s well-known that different point processes lead to different Voronoi diagrams and that Voronoi cells are a source of interesting statistics. For instance, Taylor et al. (2012) use variance in the area of Voronoi cells as a diagnostic. The result by Goldman (2014) that the authors cite could be a good starting point for a new methodology, but one would have to show a) In what way it is different from/better than classical ways of assessing repulsion, i.e. the pair correlation function, Ripley’s k-function, nearest-neighbour statistics (also related to Voronoi cells), etc. b) In particular, repulsion occurs at different spatial scales, and some processes exhibit attraction at small scales and repulsion at large ones, or vice versa (see cluster processes). How is this reflected in the typical shape of a Voronoi cell? c) Can we deal with non-homogeneous intensities and boundary effects in a more elegant way than just by windowing the data? This can be done for the pair correlation function via intensity estimation.
What is the point of the classifier here? Is it to develop a statistical test
for repulsiveness? Why not make a test then, and study it as such? In addition, the authors train the classifier is trained on many redundant features: see the sentence “we add the mean of the first 5 … cells’ areas in order to have a more global information”. Adding the mean of a set of features just gives you a redundant feature (it’s a linear function of the others), so this behaviour is weird. It also points to the fact that it’s just the average shape that matters, not the distribution of the shapes of 10 individual cells. My impression is that the authors can just a one-dimensional summary of the average circularity of the Voronoi cells and the classifier just becomes a statistical test.
Application to the cellular towers datasets. I fail to see the point here.
Obviously there’s a need to spread these cells over space so it’s not going to be a Poisson process. What does the analysis tell us we don’t know already?

In addition I see two more minor issues. My impression is that the “Preliminaries” section is needlessly technical given the rest of the paper, which doesn’t really need Janossy measures etc. The other issue is that the paper uses a notebook format which doesn’t seem well-suited to the material. Everything the authors describe is very easy to implement, so it’s not clear why the code should be part of the paper. 
[...]

References

Taylor, D. B., Dhillon, H. S., Novlan, T. D., & Andrews, J. G. (2012, December). Pairwise interaction processes for modeling cellular network topology. In 2012 IEEE Global Communications Conference (GLOBECOM) (pp. 4524-4529). IEEE. Stoyan, D., Kendall, W. S., Chiu, S. N., & Mecke, J. (2013). Stochastic geometry and its applications. John Wiley & Sons.

### 2nd review 

The paper addresses the problem of identifying repulsive configuration using a machine learning approach. Both approaches are based on the calculation of geometric quantities related to the Voronoi tessellation of the configuration. The proposed algorithm is based on some existing results on the Voronoi cells of point processes: it was shown in [6] that the Voronoi cells of a Ginibre point process are more circular than those of a Poisson point process. The paper proposes to use the ratio of the surface to the squared perimeter to quantify the ‘circularity’ of the cells. In particular, it proposes to build a feature map for each configuration, where the features correspond to the ratio of the surface to the squared perimeter, calculated for every cell. Another feature map is built by calculating the same ratio but for only ‘ the five most central cells’. These feature maps were used as inputs of logistic regression classifiers. This approach was tested on synthetic datasets obtained by simulating Ginibre point process, thinned Ginibre point process and Poisson point process. The thinning of the Ginibre point process was conducted to obtain a different point process but with lower repulsion. This approach was shown to yield an accuracy of 70% when the first feature map is used, and an accuracy of 85% when the second feature map is used. Finally, the proposed approach was tested on a real dataset called Cartoradio, which contains  configurations of some mobile phone base stations in Paris.

Overall, the paper has undergone substantial improvement since the previous submission, and the contribution is now clearer. In the following I give some remarks/suggestions to further improve the clarity and self-consistency of the paper:

- A crucial step of the proposed algorithm is the calculation of the areas and the perimeter of each cell of Voronoi tessellations. There is no explicit information on the practical calculation of these quantities. I recommend providing more details on this step.

- The selection of hyperparameters was overlooked: the decision to choose five central cells appears arbitrary. I recommend incorporating a section that explains the rationale behind choosing the hyperparameter 'number of central cells.

## Author's answer:

We would like to thank the referees for their thorough revision and the useful comments
they made. The main points raised by the referees were the following :
- Adding a formal description of the algorithm
- More comments of the Python code
- Better presentation of the manuscript
- More thorough numerical illustrations to fully leverage the submission format

We have shortened greatly the mathematical preliminairies on point processes to keep
only the necessary part useful for the comprehension of the sequel. The Python code
is now split in small pieces, each of which is intertwined with the text as the overall
procedure is described. We hopefully improved the presentation of the manuscript and
add some numerical illustrations.
We also made a comparison with an older paper [1] which was indicated by one of the
referee to motivate our approach based on Machine Learning. We found that considering
the ratio S/P 2 (surface over squared perimeter) of Voronoi cells instead of the surface
only as in [1] yields a much more robust indicator of repulsivity.
We hope that this new version will satisfy the requirements of Computo.

References
[1] D. B. Taylor, H. S. Dhillon, T. D. Novlan, and J. G. Andrews, Pairwise interaction
processes for modeling cellular network topology, Globecom, 2012.
